client/src/pages/add-recipe/add-recipe-manual.page.js

```diff
@@ -15,14 +20,71 @@ const ManualInsertPage = () => {
     const navigate = useNavigate();
     const scrollOffset = -1 * window.innerHeight * 0.1;
 
+    // CHATGPT --->
+    const [ingredients, setIngredients] = useState([{ value: "" }]);
+    const [steps, setSteps] = useState([{ value: "" }]);
+
+    const handleIngChange = (index, event) => {
+      const values = [...ingredients];
+      values[index].value = event.target.value;
+      setIngredients(values);
+    };
+    const handleStepChange = (index, event) => {
+        const values = [...steps];
+        values[index].value = event.target.value;
+        setSteps(values);
+      };
+  
+    const handleAddIng = () => {
+      const values = [...ingredients];
+      values.push({ value: "" });
+      setIngredients(values);
+    };
+    const handleAddStep = () => {
+        const values = [...steps];
+        values.push({ value: "" });
+        setSteps(values);
+      };
+  
+    const handleRemoveIng = (index) => {
+      const values = [...ingredients];
+      values.splice(index, 1);
+      setIngredients(values);
+    };
+    const handleRemoveStep = (index) => {
+      const values = [...steps];
+      values.splice(index, 1);
+      setSteps(values);
+    };
+
+    // <--- CHATGPT 
+
     const handleSubmit = (event) => {
         event.preventDefault();
         const data = new FormData(event.currentTarget);
+        
+        console.log(ingredients);
+        console.log(steps);
+        
+        // need to handle ingredients, steps
+        let ing_string = ""
+        for (let ing of ingredients) {
```

Comment:
The expected format of the ingredients going to the backend is an array of ingredient strings. See models/recipe.model.js for more information. Also, you may want to implement some code to check that ingredients aren't proceeded with a number (i.e. "1. 2 cups of flour")

Reply:
Appreciate all the notes Max! I fixed the json output so it sends arrays of ingredients and steps. I'm not too worried about ingredients being preceded by a number...I've never seen ingredients with that format I think because of how confusing it is. But let me know what you think and I can add that. I'd also be concerned about a legitimate ingredient like "2 16 oz cans of beans" being ruled out by the double number.